pkg/api/controller_test.go

```diff
@@ -3083,3 +3083,90 @@ func TestController_GetPhysicalAddress_(t *testing.T) {
 		}
 	})
 }
+
+func TestController_PrepareGarbageCollectionUncommitted(t *testing.T) {
+	clt, deps := setupClientWithAdmin(t)
+	ctx := context.Background()
+
+	t.Run("no_repository", func(t *testing.T) {
+		resp, err := clt.PrepareGarbageCollectionUncommittedWithResponse(ctx, "", api.PrepareGarbageCollectionUncommittedJSONRequestBody{})
+		if err != nil {
+			t.Fatalf("PrepareGarbageCollectionUncommittedWithResponse failed: %s", err)
+		}
+		if resp.JSON400 == nil {
+			t.Fatalf("PrepareGarbageCollectionUncommittedWithResponse expected BadRequest: %+v", resp)
+		}
+	})
+
+	t.Run("repository_not_exists", func(t *testing.T) {
+		repo := testUniqueRepoName()
+		resp, err := clt.PrepareGarbageCollectionUncommittedWithResponse(ctx, repo, api.PrepareGarbageCollectionUncommittedJSONRequestBody{})
+		if err != nil {
+			t.Fatalf("PrepareGarbageCollectionUncommittedWithResponse failed: %s", err)
+		}
+		if resp.JSON404 == nil {
+			t.Fatalf("PrepareGarbageCollectionUncommittedWithResponse expected NotFound: %+v", resp)
+		}
+	})
+
+	t.Run("uncommitted", func(t *testing.T) {
+		repo := testUniqueRepoName()
+		_, err := deps.catalog.CreateRepository(ctx, repo, onBlock(deps, repo), "main")
+		testutil.Must(t, err)
+		const items = 3
+		for i := 0; i < items; i++ {
+			path := fmt.Sprintf("uncommitted/obj%d", i)
+			uploadResp, err := uploadObjectHelper(t, ctx, clt, path, strings.NewReader(path), repo, "main")
+			verifyResponseOK(t, uploadResp, err)
+		}
+		var continuationToken *string
+		for {
+			resp, err := clt.PrepareGarbageCollectionUncommittedWithResponse(ctx, repo, api.PrepareGarbageCollectionUncommittedJSONRequestBody{
+				ContinuationToken: continuationToken,
+			})
+			verifyResponseOK(t, resp, err)
+			if resp.JSON201.RunId == "" {
+				t.Errorf("PrepareGarbageCollectionUncommittedWithResponse empty RunID, value expected")
+			}
+			if resp.JSON201.Location == "" {
+				t.Errorf("PrepareGarbageCollectionUncommittedWithResponse empty Location, value expected")
+			}
+			token := api.StringValue(resp.JSON201.ContinuationToken)
```

Comment:
We should validate that token == "" for items = 3, it is a bug if we get a different result

Reply:
Endless loop, that will cause timeout - but I'll add verification for both committed and uncommitted.